src/wallet/rpcwallet.cpp

```diff
@@ -2361,19 +2361,18 @@ UniValue lockunspent(const JSONRPCRequest& request)
 
     LOCK2(cs_main, pwallet->cs_wallet);
 
-    if (request.params.size() == 1)
-        RPCTypeCheck(request.params, {UniValue::VBOOL});
-    else
-        RPCTypeCheck(request.params, {UniValue::VBOOL, UniValue::VARR});
+    RPCTypeCheckArgument(request.params[0], UniValue::VBOOL);
 
     bool fUnlock = request.params[0].get_bool();
 
-    if (request.params.size() == 1) {
+    if (request.params[1].isNull()) {
```

Comment:
Thread https://github.com/bitcoin/bitcoin/pull/11050#discussion_r133097936

No, params[1] is the `transactions` argument to the `lockunspent` RPC. The `transactions` argument may be an array of transaction outputs, or it may be null, or it may be missing.

This code is just saying that if the `transactions` argument is null or missing, then all transactions should be unlocked. The previous code only worked if `transactions` argument was missing, but not if the `transactions` argument was null (because it would trigger a type error).